src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt

```diff
@@ -157,18 +158,26 @@ class AppointmentService(
       return
     }
 
-    if (!deleteRelatedAppointments) {
-      prisonApiService.deleteAppointment(appointmentId)
-      removeSingleAppointmentInRecurringList(appointmentId, recurringAppointment!!)
-      return
+    prisonApiService.deleteAppointment(appointmentId)
+    removeSingleAppointmentInRecurringList(appointmentId, recurringAppointment)
+    return
+  }
+
+  @Transactional
+  fun deleteRecurringAppointmentSequence(recurringAppointmentId: Long) {
+    val recurringAppointment: Optional<RecurringAppointment> =
+      recurringAppointmentRepository.findById(recurringAppointmentId)
```

Comment:
Could you do a .orElseThrow here? It could get rid of the if statement and the recurringAppointment.get() lower down.

Reply:
That's the one - I'll fix